front/src/components/boxs/chart-feature/EditChartOneFeatures.jsx

```diff
@@ -0,0 +1,144 @@
+import { Component } from 'preact';
+import { Text } from 'preact-i18n';
+import { connect } from 'unistore/preact';
+import Select from 'react-select';
+
+import BaseEditBox from '../baseEditBox';
+import ChartTypeSelector from './ChartTypeSelector';
+import ChartPeriodSelector from './ChartPeriodSelector';
+import { getDeviceFeatureName } from '../../../utils/device';
+import { DEVICE_FEATURE_TYPES } from '../../../../../server/utils/constants';
+
+import actions from '../../../actions/dashboard/edit-boxes/editChartFeatures';
+
+@connect('httpClient', actions)
+class EditChartMultiFeatures extends Component {
+  updateBoxChartType = chartType => {
+    this.props.updateBoxChartType(this.props.x, this.props.y, chartType.value);
+  };
+
+  updateBoxChartPeriod = chartPeriod => {
+    this.props.updateBoxChartPeriod(this.props.x, this.props.y, chartPeriod.value);
+  };
+
+  updateDeviceFeatures = selectedDeviceFeaturesOptions => {
+    selectedDeviceFeaturesOptions = selectedDeviceFeaturesOptions || {};
+    const deviceFeatures = [selectedDeviceFeaturesOptions.value];
+    this.props.updateBoxDeviceFeatures(this.props.x, this.props.y, deviceFeatures);
+    this.setState({ selectedDeviceFeaturesOptions });
+  };
+
+  getDeviceFeatures = async () => {
+    try {
+      this.setState({ loading: true });
+
+      const excludeFeatyreType = [
```

Comment:
typo

Reply:
sorry, I don't understand what is typo pb ?

Reply:
excludeFeatyreType => excludeFeatureType

Reply:
I fix this